File: 125-ValidPalindrome/125-ValidPalindrome.java
// Last updated: 8/19/2026, 10:40:41 AM
class Solution {
    public boolean isPalindrome(String s) {
        s=s.toLowerCase();
        s=s.replaceAll("[^a-z0-9]", "");
        int left=0,right=s.length()-1;
        while(left<right){
            if(s.charAt(left)!=s.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}

File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 8/19/2026, 10:40:46 AM
class Solution:
    def isPalindrome(self, s: str) -> bool:
        symbols = 'abcdefghijklmnopqrstuvwxyz1234567890'
        result = ''

        for i in s.lower():
            if i in symbols:
                result += i
        
        return result[::-1] == result